zerver/data_import/mattermost.py

```diff
@@ -0,0 +1,540 @@
+        mention_map[message_id] = mention_user_ids
+
+        content = fix_mentions(
+            content=raw_message['content'],
+            mention_user_ids=mention_user_ids,
+        )
+        content = h.handle(content)
+
+        if len(content) > 10000:
+            logging.info('skipping too-long message of length %s' % (len(content),))
+            continue
+
+        pub_date = raw_message['pub_date']
+        try:
+            recipient_id = get_recipient_id(raw_message)
+        except KeyError:
+            logging.debug("Could not find recipient_id for a message, skipping.")
+            continue
+
+        rendered_content = None
+
+        topic_name = 'imported from mattermost'
+        user_id = raw_message['sender_id']
+
+        message = build_message(
+            content=content,
+            message_id=message_id,
+            pub_date=pub_date,
+            recipient_id=recipient_id,
+            rendered_content=rendered_content,
+            topic_name=topic_name,
+            user_id=user_id,
+            has_attachment=False,
+        )
+        zerver_message.append(message)
+
+    zerver_usermessage = make_user_messages(
+        zerver_message=zerver_message,
+        subscriber_map=subscriber_map,
+        is_pm_data=is_pm_data,
+        mention_map=mention_map,
+    )
+
+    message_json = dict(
+        zerver_message=zerver_message,
+        zerver_usermessage=zerver_usermessage,
+    )
+
+    dump_file_id = NEXT_ID('dump_file_id' + str(realm_id))
+    message_file = "/messages-%06d.json" % (dump_file_id,)
+    create_converted_data_files(message_json, output_dir, message_file)
+
+def process_posts(team_name: str,
+                  realm_id: int,
+                  post_data: List[Dict[str, Any]],
+                  get_recipient_id: Callable[[ZerverFieldsT], int],
+                  subscriber_map: Dict[int, Set[int]],
+                  output_dir: str,
+                  is_pm_data: bool,
+                  masking_content: bool,
+                  user_id_mapper: IdMapper,
+                  user_handler: UserHandler,
+                  username_to_user: Dict[str, Dict[str, Any]]) -> None:
+
+    post_data_list = [
+        d
+        for d in post_data
+        if d["team"] == team_name
+    ]
+
+    def message_to_dict(post_dict: Dict[str, Any]) -> Dict[str, Any]:
+        sender_id = user_id_mapper.get(post_dict["user"])
+        content = post_dict['message']
+
+        if masking_content:
+            content = re.sub('[a-z]', 'x', content)
+            content = re.sub('[A-Z]', 'X', content)
+
+        return dict(
+            sender_id=sender_id,
+            receiver_id=post_dict["channel"],
+            content=content,
+            pub_date=int(post_dict['create_at'] / 1000),
+        )
+
+    raw_messages = []
+    for post_dict in post_data_list:
+        raw_messages.append(message_to_dict(post_dict))
+        message_replies = post_dict["replies"]
+        if message_replies is not None:
```

Comment:
I'd add a comment here explaining that Mattermost's "replies" threading feature places the additional "reply" messages on the main message object.  And that for now, we just append them immediately after the original message.